File: soccer/src/main/java/com/soccer_stats/soccer/model/Player.java
package com.soccer_stats.soccer.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "players")
@Entity
public class Player {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String firstName;
    private String lastName;
    private Position position;
    private Date birthDate;
    private double height;
    private double weight;
    private double marketValue;
    private Date contractEndDate;

    @ElementCollection(fetch = FetchType.LAZY)
    @CollectionTable(name = "player_positions", joinColumns = @JoinColumn(name = "player_id"))
    @Enumerated(EnumType.STRING)
    private Set<Position> positions;

    @ManyToOne
    @JoinColumn(name = "team_id")
    private Team currentTeam;

    @OneToMany(mappedBy = "player", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<PlayerStats> stats;
}

File: search-service-kt/src/main/kotlin/com/soccer_stats/search_service_kt/document/Player.kt
package com.soccer_stats.search_service_kt.document

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document

@Document(indexName = "playerss")
class Player(
    @Id val id: String? = null,
    val firstName: String,
    val lastName: String,
    val position: String,
    val height: Double,
    val weight: Double,
    val marketValue: Double,
    val positions: Set<String>,
    val currentTeamId: Int

) {
}